lib/Parser/parse.cpp

```diff
@@ -1286,6 +1295,29 @@ ParseNodePtr Parser::CreateNodeWithScanner(charcount_t ichMin)
     return CreateNodeT<nop>(ichMin, m_pscan->IchLimTok());
 }
 
+ParseNodePtr Parser::CreateProgNodeWithScanner(bool isModuleSource)
+{
+    ParseNodePtr pnodeProg;
+
+    if (isModuleSource)
+    {
+        pnodeProg = CreateNodeWithScanner<knopModule>();
+
+        // knopModule is not actually handled anywhere since we would need to handle it everywhere we could 
+        // have knopProg and it would be treated exactly the same except for import/export statements.
+        // We are only using it as a way to get the correct size for PnModule.
+        // Consider: Should we add a flag to PnProg which is false but set to true in PnModule? 
+        //           If we do, it can't be a virtual method since the parse nodes are all in a union.
```

Comment:
I would also argue that it can't be virtual because even if the parse nodes were made a proper class hierarchy instead of a union thing that hierarchy should be non-virtual to remain lean (virtualizing the hierarchy would add vtables to all node objects). I have a long back logged dream of refactoring the parse nodes into a proper hierarchy and doing away with the union and manual sizing and initialization craziness.  Never time for doing it though.